### ISSUE TITLE
Fails to run on macOS: `Vulkan support is either not configured in SDL or not available in current SDL video driver (cocoa) or platform`

### ISSUE THREAD

**Issue by kuka0len** [Kethku/neovide#453]
When I run `./neovide` I get this error:

```output
thread 'main' panicked at 'Failed to create window: SdlError("Vulkan support is either not configured in SDL or not available in current SDL video driver (cocoa) or platform")', src/window/sdl2/mod.rs:405:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by LakeByTheWoods** [comment]
See https://github.com/Kethku/neovide/issues/368#issuecomment-704650261

Looks like the most recent build of sdl2 was built without vulkan support on MacOS

**Comment by kuka0len** [comment]
It worked, thanks.
I just had downgrade `sdl2-sys` form `0.34.3` to `0.34.2`.

**Comment by nerosnm** [comment]
I'm having this issue on macOS 10.15.7 when building from the latest `main` (5c6d84a).

**Comment by Kethku** [comment]
@nerosnm could you try with current main? I swapped to opengl.

**Comment by nerosnm** [comment]
@Kethku It does work now, although it prints:

```
Ignored client type property: "methods"
Ignored client type property: "attributes"
```

I also needed to build using `cargo +nightly build --release` instead of `cargo build --release`, because of:

```
error: failed to parse manifest at `/Users/soren/src/github.com/Kethku/neovide/Cargo.toml`

Caused by:
  feature `resolver` is required

  this Cargo does not support nightly features, but if you
  switch to nightly channel you can add
  `cargo-features = ["resolver"]` to enable this feature
```

Maybe adding a `rust-toolchain` file would be helpful?

**Comment by jdu** [comment]
I'm having the same issue on MacOS Big Sur. On latest main and was having it previously a few weeks ago (never resolved it). 

Building in release mode:

```
    Finished release [optimized + debuginfo] target(s) in 2.66s
➜  neovide git:(main) ./target/release/neovide
Ignored client type property: "methods"
Ignored client type property: "attributes"
➜  neovide git:(main) thread 'main' panicked at 'Failed to create window: SdlError("Vulkan support is either not configured in SDL or not available in current SDL video driver (cocoa) or platform")', src/window/sdl2/mod.rs:442:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by jdu** [comment]
The opengl branch builds and runs on macos Big Sur 11.2.3

**Comment by jdu** [comment]
<img width="2605" alt="Screenshot 2021-03-23 at 21 40 01" src="https://user-images.githubusercontent.com/120225/112222206-5520ff00-8c20-11eb-97bd-1a199e1b66d6.png">

Though it's lookin a little rough...

**Comment by a06kin** [comment]
Current master with `sdl2-sys = 0.34.3` doesn't for me (Big Sur 11.3 Beta)
I just tried to update `sdl2-sys` to `0.34.4` and it worked.
Seems like `0.34.3` really don't like macos.

I build it from last master simply with `cargo build --release`

**Comment by i3d** [comment]
> Current main with `sdl2-sys = 0.34.3` doesn't for me (Big Sur 11.3 Beta)
> I just tried to update `sdl2-sys` to `0.34.4` and it worked.
> Seems like `0.34.3` really don't like macos.
> 
> I build it from last main simply with `cargo build --release`

Do you have a detail step how to update sdl2-sys to 0.34.4?

I tried something obvious but it seems wrong... So what I did was to change the Cargo specs to 0.34.4, but I got this

cargo build --release
error: failed to select a version for `libc`.
    ... required by package `sdl2 v0.34.4`
    ... which is depended on by `skulpin-renderer-sdl2 v0.5.0 (https://github.com/aclysma/skulpin?branch=master#8a913ed5)`
    ... which is depended on by `skulpin v0.11.1 (https://github.com/aclysma/skulpin?branch=master#8a913ed5)`
    ... which is depended on by `neovide v0.7.0 (/Users/jimxu/src/neovide)`
versions that meet the requirements `^0.2.92` are: 0.2.93, 0.2.92

all possible versions conflict with previously selected packages.

  previously selected package `libc v0.2.82`
    ... which is depended on by `atty v0.2.14`
    ... which is depended on by `clap v2.33.3`
    ... which is depended on by `bindgen v0.54.0`
    ... which is depended on by `skia-bindings v0.32.1`
    ... which is depended on by `skia-safe v0.32.1`
    ... which is depended on by `neovide v0.7.0 (/Users/jimxu/src/neovide)`

failed to select a version for `libc` which could resolve this conflict

**Comment by i3d** [comment]
ah sorry, looks like hand editing isn't the way to do, need to use the `cargo update -p sdl2-sys --precise 0.34.4` command.

**Comment by Kethku** [comment]
I just merged a PR which bumped sdl2 to 0.34.4. Hopefully that fixes the issue

**Comment by holicc** [comment]
> I just merged a PR which bumped sdl2 to 0.34.4. Hopefully that fixes the issue

not work for me.  

MacOS: 10.15.7 (19H1217)

**Comment by Kethku** [comment]
I think this has been fixed in main. Please reopen if you come back to this and the issue remains.
